packages/service/lib/plugins/sandbox-wrapper.js

```diff
@@ -33,3 +43,47 @@ module.exports = fp(async function (app, opts) {
     }
   }
 })
+
+/**
+ * Creates an object for pattern specific options. This ensures that
+ * only configurations that have been provided are included in the
+ * final result. This prevents 'cannot read properties of undefined'
+ * errors when undefined configs are provided to the underlying
+ * @fastify/autoload plugin.
+ */
+function patternOptionsFromPlugin (plugin) {
+  const config = {}
+
+  // Expected keys for autoload plugin options that expect regexp patterns
+  const patternOptionKeys = ['ignorePattern', 'indexPattern', 'autoHooksPattern']
+
+  for (const key of patternOptionKeys) {
+    const pattern = plugin[key]
+
+    // If pattern key not found in plugin object, move on
+    if (!pattern) {
+      continue
+    }
+
+    // Build an instance of a RegExp. If this comes back undefined,
+    // then an invalid value was provided. Move on.
+    const regExpPattern = stringPatternToRegExp(pattern)
+    if (!regExpPattern) {
+      continue
+    }
+
+    // We have a valid RegExp so add the option to the config to pass along to
+    // autoload.
+    config[key] = regExpPattern
+  }
+
+  return config
+}
```

Comment:
Ah, good thing for that code coverage check! I was previously setting pattern specific properties to undefined if a value did not exist but then realized it was causing the `autoload` plugin to explode downstream. So I changed it such that we only set the RegExp if it's valid otherwise the key is not included in the final configuration.